### PR TITLE
Better loading of internal and user functions.

### PR DESCRIPTION
**Pull request by tobiasraabe**
### What problem do you want to solve?

- Better error messages when functions with the same name are loaded.
- Internal functions are loaded with `load_functions("gettsim.soz_vers.krankenv_pflegev")` so that functions have `func.__module__` set which facilitates linking to the docs.
- Tests for `function_loader.py`.

### Todo

- [x] Re-check with implementation in #184.

### CODE REVIEW

File: gettsim/config.py [hunk 1]
```diff
@@ -7,10 +7,10 @@
 GEP_1_CHARACTER_LIMIT = 17
 
 INTERNAL_FUNCTION_FILES = [
```
Comment (hmgaudecker) [inline gettsim/config.py:9]: `INTERNAL_FUNCTION_LOCATIONS` ? Most of these are directories, not files.

File: docs/tutorials/user_functions.ipynb [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+    "user_functions"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Modules\n",
+    "\n",
+    "You can also pass a module as a user function. Assume you have import `abgelt_st.py`, you can pass all of its functions as user functions with"
```
Comment (hmgaudecker) [inline docs/tutorials/user_functions.ipynb:323]:
```suggestion
    "You can also pass a module name; all functions in this module will be loaded. Assume you have import `abgelt_st.py`, you can pass all of its functions as user functions with"
```

File: gettsim/config.py (hunk 1 above)
Comment (tobiasraabe) [reply gettsim/config.py:9]: Renaming to `PATHS_TO_INTERNAL_FUNCTIONS`?

Comment (hmgaudecker) [reply gettsim/config.py]: Fine with me. Maybe a little note that this will take all modules in a directory or a single module?

**Comment by hmgaudecker** [comment]
> If you do not think it is necessary to have it in 0.3.x, I would merge it with 0.4 and update the tutorial when another PR establishes the new functions names.

The only necessary thing are bugfixes in the taxes & transfers; this does seem orthogonal.

**Pull request merged**